### PR TITLE
Add bdd-style test framework

### PR DESCRIPTION
**Pull request by benedictvh**
I force pushed your changes to my origin like you showed me, but before applying my own changes, and I guess that made github close the original pull request automatically.   Creating a second pull request for the same content seems unnecessary, but I was not sure how to reopen the original one.

Anyway, I think I have resolved all the issues you put forward.

There are still some pylint warnings about missing docstrings, which I am guessing you are comfortable with ignoring, and some other warnings about redefining outer scope variables, which I cannot get rid of because pytest-bdd finds dependencies that way.

### CODE REVIEW

File: tests/test_spread.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from pytest import fixture
+from pytest_bdd import scenario, given, when, then, parsers
+
+from beancount.core.compare import hash_entry, includes_entries
+from beancount.loader import load_string
+from beancount.parser import printer
+
+from .context import spread
```
Comment (Akuukis) [inline tests/test_spread.py]: Did it work for you? It doesn't work for me with Python 3.7.5 and requires this change

```suggestion
from context import spread
```

File: tests/test_spread.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    # Get correctly generated transactions from feature file
+    correctly_generated_txns, _, _ = load_string(correctly_generated_txn_text)
+
+    has_correct, missing_entries = includes_entries(correctly_generated_txns, generated_txns)
+    assert has_correct
+    printer.print_entries(missing_entries)
+
+    # Only 4 of the correct entries are written in feature file for the sake of brevity
+    # Make sure that there are the right number generated by the plugin
+    assert len(generated_txns) == 30
```
Comment (Akuukis) [inline tests/test_spread.py]: Good catch to check the amount of generated txns! But don't hardcode this number, instead create a new `@then` along the lines of "should generate X transactions".

**Review by Akuukis — CHANGES_REQUESTED** [review]
Looks good! Please see two comments in the review and then it's good to merge

File: tests/test_spread.py (hunk 1 above)
Comment (benedictvh) [reply tests/test_spread.py]: oh ok, no it didn't work for me, I thought it was some kind of problem with my own installation or something.

File: tests/test_spread.py (hunk 2 above)
Comment (benedictvh) [reply tests/test_spread.py]: Much better idea. Thanks.

**Comment by Akuukis** [comment]
Thanks a lot, merged!

**Pull request merged**